### PR TITLE
Update getNodeExporterNodes to use Prometheus

### LINKED ISSUE #15: Don't hardcode prom-exporter=node label selector for prometheus backend
#### What is the current behavior?
See these discussions:

- https://github.com/containership/cerebral/pull/13#discussion_r238404175
- https://github.com/containership/cerebral/pull/13#discussion_r238404797

#### What is the expected behavior?

We shouldn't hardcode the label selector as it's specific to our setup. Let's see if we can refactor that block of code to use a dynamic label selector built from the Prometheus `targets` response (can be retrieved using `prometheusclient.Targets()`. Or maybe we can somehow grab the `PodIP`s we need directly from the targets.

#### Additional Context

Pulling this into the release so we can try to get it in, but I don't think it should block the release.

### PR DESCRIPTION
**Pull request by noah-vienneau**
### Description

 #### What does this pull request accomplish?
Eliminates the need to use a label selector in favor of the prometheus Targets api call, which gets rid of needing to hard code "prom-exporter". All changes made to the getNodeExporterPodIPsOnNodes function.  
 #### What issue(s) does this fix?
* Fixes #15 

 ### Testing
No changes to the existing tests
Only tested using Unit Tests

### CODE REVIEW

File: pkg/metrics/backends/prometheus/prometheus.go [hunk 1]
```diff
@@ -127,25 +125,16 @@ func (b Backend) GetValue(metric string, configuration map[string]string, nodeSe
...
-	// TODO we should rethink how we do all of the below, potentially using b.Targets()
-	// to help. See https://github.com/containership/cerebral/issues/15.
-	selector := labels.NewSelector()
-	l, _ := labels.NewRequirement("prom-exporter", selection.Equals, []string{"node"})
-	selector = selector.Add(*l)
-
-	// TODO if we could use a FieldSelector including Pod.Spec.NodeName then we
-	// could avoid the gross O(n^2) below. Is there a way to do that without
-	// having to query the API directly?
-	pods, err := b.podLister.List(selector)
```
Comment (mattkelly) [inline pkg/metrics/backends/prometheus/prometheus.go]: I believe we can remove the `podLister` from this backend entirely now, right? That would be great if possible.

File: pkg/metrics/backends/prometheus/prometheus.go [hunk 2]
```diff
@@ -4,18 +4,16 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/pkg/errors"
+	"strings"
 	"text/template"
 	"time"
 
-	"github.com/pkg/errors"
```
[the remark this answers is not in the document]
Comment (noah-vienneau) [reply pkg/metrics/backends/prometheus/prometheus.go]: Will do thanks for letting me know

File: pkg/metrics/backends/prometheus/prometheus.go [hunk 3]
```diff
@@ -127,25 +125,16 @@ func (b Backend) GetValue(metric string, configuration map[string]string, nodeSe
...
-	if err != nil {
-		return nil, errors.Wrap(err, "listing node exporter pods")
-	}
-	// Only filter in prom-exporter pods that belong to a node we care about
-	for _, pod := range pods {
-		for _, node := range nodes {
-			if pod.Spec.NodeName == node.ObjectMeta.Name {
-				podIPs = append(podIPs, pod.Status.PodIP)
+	//Filter only prom-exporter job, furhter filter down by node ips
+	targets, _ := b.prometheus.Targets(context.TODO())
```
[the remark this answers is not in the document]
Comment (noah-vienneau) [reply pkg/metrics/backends/prometheus/prometheus.go]: Ok, I was unsure what context to use. That makes sense

File: pkg/metrics/backends/prometheus/prometheus.go (hunk 1 above)
Comment (noah-vienneau) [reply pkg/metrics/backends/prometheus/prometheus.go]: Looks like it is possible
Just changed in next commit

Comment (noah-vienneau) [reply pkg/metrics/backends/prometheus/prometheus.go]: Yes removing it and a call to newClient in metrics_backend which requires a podLister argument works

**Comment by mattkelly** [comment]
Thanks so much for taking this on, @noah-vienneau! Nice work. One of us from @containership/core-services will handle testing this on a real cluster. 🚀

File: pkg/metrics/backends/prometheus/prometheus.go [hunk 4]
```diff
@@ -41,6 +39,9 @@ const cpuQueryTemplateString = `
 	) * 100
 )`
 
+
```
Comment (mattkelly) [inline pkg/metrics/backends/prometheus/prometheus.go]: Remove this change

**Pull request merged**